# Copyright (c) 2019, NVIDIA CORPORATION.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

# Treelite requires protobuf during the build process. It generaly uses the
# static protobuf present in its thirdparty folder. This patch enables
# treelite to build with the protobuf installed using conda and not
# the one present in its thirdparty folder.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b07f9c6..108603a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,12 +22,8 @@ set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
 # Protobuf
 if (ENABLE_PROTOBUF)
-  set(protobuf_BUILD_TESTS OFF CACHE BOOL "enable tests for protobuf" FORCE)
-  set(protobuf_BUILD_SHARED_LIBS OFF CACHE BOOL "enable shared libs for protobuf" FORCE)
-  add_subdirectory(3rdparty/protobuf/cmake)
-  set(PROTOBUF_PROTOC_EXECUTABLE protoc)
-  set(PROTOBUF_INCLUDE_DIRS "${PROJECT_SOURCE_DIR}/3rdparty/protobuf/src")
-  set(PROTOBUF_LIBRARIES libprotobuf)
+  set(Protobuf_USE_STATIC_LIBS ON)
+  find_package(Protobuf REQUIRED)
   add_definitions(-DTREELITE_PROTOBUF_SUPPORT)
 else()
   set(PROTOBUF_LIBRARIES "")